Angular/src/app/add-role-modal/add-role-modal.component.html

```diff
@@ -0,0 +1,86 @@
+<div class="modal-header">
+  <h5 class="modal-title" id="exampleModalLabel">Add Role</h5>
+  <button
+    type="button"
+    class="close"
+    aria-label="Close"
+    (click)="activeModal.dismiss()"
+  >
+    x
+  </button>
+</div>
+
+<form [formGroup]="addRoleForm" (ngSubmit)="onSubmit()" autocomplete="off">
+  <div class="modal-body" data-keyboard="false" data-backdrop="static">
+    <div class="form-group">
+      <label for="roleName">Role Name</label>
+      <input
+        type="text"
+        formControlName="roleName"
+        class="form-control"
+        [ngClass]="{ 'is-invalid': submitted && formControls.roleName.errors }"
+      />
+      <div
+        *ngIf="submitted && formControls.roleName.errors"
+        class="invalid-feedback"
+      >
+        <div *ngIf="formControls.roleName.errors.required">
+          Role Name is a required field
+        </div>
+      </div>
+    </div>
+
+    <div class="form-group">
```

Comment:
Make all fields in this form required, even though some of the current data from the career lattice is missing. These should be required to ensure that empty role information isn't added.

Reply:
sorted